File: src/main/scala/xf/examples/JsonResponse.scala
package xf.examples

import cats.effect.{ExitCode, IO, IOApp}
import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.*
import io.circe.*
 import xf.examples.Common.{clientResource, createTangibleClient, extractKey}

object JsonResponse extends IOApp:

  case class Person(name: String, nationality: String, age: Int)
  given Codec[Person] = deriveCodec

  override def run(args: List[String]): IO[ExitCode] =
    clientResource
      .use { client =>
        val tc      = createTangibleClient(client, extractKey())
        val example = Person("Jose", "Spain", 52)

        for
          response <- tc.expectJson("Give me 10 random people", List(example))
          _        <- response.map { r => IO.println(r.value) }.getOrElse( IO.raiseError(new RuntimeException()) )
        yield ExitCode.Success
      }

File: typescript-experiment/src/examples/JSONResponse.ts
import { tangibleClient } from "./common.ts";

type Person = {
    name: string;
    nationality: string;
    age: number;
};

const tc = tangibleClient();

const example: Person = { name: "Jose", nationality: "Spain", age: 52 };
const response = await tc.expectJson("Give me 10 random people", [example]);

if (response.outcome === "Success") {
    console.log(response.value);
}